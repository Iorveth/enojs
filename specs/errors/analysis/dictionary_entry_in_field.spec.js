const eno = require('../../../eno.js');

const input = `
language:
| eno
json = JavaScript Object Notation
`.trim();

describe('analysis.dictionaryEntryInField', () => {

  let error;

  beforeAll(() => {
    try {
      eno.parse(input);
    } catch(err) {
      error = err;
    }
  })

  it(`provides the correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toMatchSnapshot();
  });
});
