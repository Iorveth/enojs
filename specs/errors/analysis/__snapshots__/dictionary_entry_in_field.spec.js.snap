// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analysis.dictionaryEntryInField provides correct selection metadata 1`] = `
Array [
  Array [
    2,
    0,
  ],
  Array [
    2,
    33,
  ],
]
`;

exports[`analysis.dictionaryEntryInField provides the correct message 1`] = `
"Line 3 contains a dictionary entry inside a field.

   Line | Content
 *    1 | language:
 *    2 | | eno
 >    3 | json = JavaScript Object Notation
"
`;
